src/main/kotlin/stringcaclulator/Calculator.kt

```diff
@@ -0,0 +1,76 @@
+package stringcaclulator
+
+import java.lang.IllegalArgumentException
+import kotlin.math.round
+
+class Calculator {
+    fun calculate(input: String): Double {
+        val inputList = splitInput(input)
+        return calculateWithSplitList(inputList)
+    }
+
+    fun splitInput(input: String): List<String> {
+        return input
+            .split(" ")
+    }
+
+    fun calculateWithSplitList(splitList: List<String>): Double {
+        checkNumber(splitList[0])
+        var result = splitList[0].toDouble()
+
+        for (i in 1 until splitList.size step 2) {
+            val operator = splitList[i]
+            val number = splitList[i + 1]
+            result = checkThenCalculate(
+                operator,
+                result,
+                number
+            )
+        }
+        return result
+    }
+
+    fun checkThenCalculate(operator: String, result: Double, number: String): Double {
+        checkNumber(number)
+        checkOperator(operator)
+        return when (operator) {
+            "+" -> add(result, number.toDouble())
+            "-" -> subtract(result, number.toDouble())
+            "*" -> multiply(result, number.toDouble())
+            "/" -> divide(result, number.toDouble())
+            else -> throw IllegalArgumentException("연산자 위치에 잘못된 값이 들어있습니다. 잘못된 연산자: $operator")
+        }
+    }
+
+    fun checkNumber(number: String): Boolean {
+        val result = "[0-9]+|[0-9]+[.][0-9]+".toRegex()
```

Comment:
자바에서 Boolean 값에 해당하는 변수는 is~~~ 처럼 관용적으로 사용되고 있어.
```suggestion
        val isNumber = ~~~
```

Reply:
오 감사합니다! 이 부분도 반영할게요!!